### ISSUE TITLE
[Test] - JDK 1.8 / 18

### ISSUE THREAD

**Issue by GeekSheikh** [databrickslabs/overwatch#435]
Tests cannot execute in latest version of JAVA with `sbt clean assembly` due to security manager

```
java.lang.UnsupportedOperationException: The Security Manager is deprecated and will be removed in a future release
        at java.base/java.lang.System.setSecurityManager(System.java:416)
        at sbt.TrapExit$.installManager(TrapExit.scala:53)
        at sbt.StandardMain$.runManaged(Main.scala:128)
        at sbt.xMain.run(Main.scala:89)
        at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:111)
        at xsbt.boot.Launch$.withContextLoader(Launch.scala:130)
        at xsbt.boot.Launch$.run(Launch.scala:111)
        at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:37)
        at xsbt.boot.Launch$.launch(Launch.scala:119)
        at xsbt.boot.Launch$.apply(Launch.scala:20)
        at xsbt.boot.Boot$.runImpl(Boot.scala:56)
        at xsbt.boot.Boot$.main(Boot.scala:18)
        at xsbt.boot.Boot.main(Boot.scala)
[error] [launcher] error during sbt launcher: java.lang.UnsupportedOperationException: The Security Manager is deprecated and will be removed in a future release
```

**Comment by deepamsingla** [comment]
SBT recently upgraded to scala 2.12.16

**Comment by deepamsingla** [comment]
SBT recently upgraded to scala 2.12.16

**Comment by deepamsingla** [comment]
SBT recently upgraded to scala 2.12.16

https://github.com/sbt/sbt/releases/tag/v1.7.0-RC2

this version of scala is pretty old and supports only java8

https://www.scala-lang.org/news/2.12-roadmap/
https://www.infoq.com/articles/Scala-2-12-Only-Java8/

I was able to make it work only after making sbt point to java 8.

**Comment by deepamsingla** [comment]
I made an attempt to bring sbt version to 1.7.1 and it caused many code breaks the main issue has been backward compatibility. I updated the build.sbt file but then it started contradicting with the older version plugins. Again the issue was with the backward compatibility. The changed file can be found attached below.

[build.sbt.txt](https://github.com/databrickslabs/overwatch/files/9357223/build.sbt.txt)
